js/app/model.js

```diff
@@ -89,22 +72,50 @@ var dreamsModel =  {
     })
   },
 
-  getDreamsForTag: function (tag) {
+  // private
+
+  pathForFetchType: function (fetchType) {
+    if (typeof fetchType === 'object' && fetchType.tag) {
+      return '/tag/dreams?tag=' + fetchType.tag
+    } else if (fetchType === 'forUser') {
+      return '/user/dreams'
+    } else {
+      return '/dreams'
+    }
+  },
+
+  fetchViews: function () {
```

Comment:
Don't fully understand what is going on here, But I will look at it more closely tomorrow. Gotta do contracting tonight.